### ISSUE TITLE
Request: include replaced txid for `dropped_replace_by_fee`

### ISSUE THREAD

**Issue by hstove** [hirosystems/stacks-blockchain-api#1588]
When building Stacks apps, it's common to have some mechanism where you track a pending transaction ID. When a user manually bumps up the fee, it's tricky to "replace" the txid you're watching. For most apps, bumping the fee breaks whatever app you're using if they're showing a pending state. As far as I know, the way to do this would be:

- catch when status is `dropped_replace_by_fee`
- fetch both mempool and confirmed tx's for the sender
- find the tx with the same nonce as the one that was dropped

Given that the API knows what the 'canonical' (ie not dropped) tx is, it would be really helpful if something like `replacement_txid` was returned from the API for dropped transactions.

**Comment by rafaelcr** [comment]
We discussed this briefly today and we concluded that if we implement something like this on the API side, it would be a "lossy" feature due to the fact that mempool messages are best-effort and there could be cases where we don't see the new RBFd transaction in it.

A more robust solution would be for the Stacks node which emitted the RBF tx to also emit a new event of some kind that indicates which tx id it is replacing. That way, we API could index that event and return it in tx endpoints. It might make more sense to open an issue on the blockchain side so this possibility can be discussed.
